pkg/filesystem/image.go

```diff
@@ -117,11 +117,6 @@ func (fs *FileSystem) generateThumbnail(ctx context.Context, file *model.File) e
 	defer cancel()
 	// TODO: check file size
 
-	if file.Size > uint64(model.GetIntSetting("thumb_max_src_size", 31457280)) {
```

Comment:
unhappy path first - better to check max size before obtain workers.

Reply:
TLDR: I moved this code to below that local download checking, generating thumbnail shouldn't have a max size limit on.

---

So you want max size check to be the first priority?
Because by using url method size can be ignored and unlimited.

By letting size check for local (the code below) this could safely confirmes that we're not downloading huge files at local.

If this line puts in first this could break how this `generate by url` method goes.

For example,
if I increase the max size to 10GB, (e.g. Some kind of movie)
when I failed to get the source file, the logic will fallbacks to local download mode
that's the place that the max size is worth for checking to avoid downloading huge files to local vm (As mentioned above)

In another way, if you check the max size first, this could cause lots of file that over the max size can't generate the thumbnail and forced ppl to increase in settings.
If they don't have big enough hard disk in local, this could cause OOM / full hard disk problem.